src/components/Navbar/NavItem.js

```diff
@@ -0,0 +1,27 @@
+//Dependencies
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+//Components, assets, actions, styles etc..
+import { setActiveMenu } from '../../actions/index';
+
+const NavItem = (props) => {
+  const navState = useSelector((state) => state.nav);
+  const dispatch = useDispatch();
+
+  const setMenu = () => dispatch(setActiveMenu(props.name));
+  const closeMenu = () => dispatch(setActiveMenu(false));
+  const setClass = () => (props.name === 'Donate' ? 'donate' : 'nav');
```

Comment:
This seems like over kill with using redux - instead i think it makes more sense to just keep this to component state.

Reply:
Example: 
```
const NavItem = (props) => {
  const [ navState, setNavState ] = useState(null);

  const setMenu = () => setNavState(props.name);
  const closeMenu = () => setNavState(null);
  const setClass = () => (props.name === 'Donate' ? 'donate' : 'nav');

  return (
    <div className={`${setClass()}--link`} onMouseEnter={setMenu} onMouseLeave={closeMenu}>
      <Link className={`${setClass()}--link__txt`} to={props.linkTo}>
        {props.name}
      </Link>
      {navState && props.children}
    </div>
  );
};
```

Reply:
This way we don't do extra calls to the store - we dont need to mutate state here so we shouldnt be using the store to handle this state.

Reply:
☝️ this is a okay solution intention is to just to show functionality

Reply:
Navbar.js has been refactored to hold component level state via react hooks and avoid calling redux store for conditional rendering of dropdown menu.